bridge/client/app/project-board/project-board.component.ts

```diff
@@ -228,12 +228,12 @@ export class ProjectBoardComponent implements OnInit, OnDestroy {
     this.selectedStage = stage;
   }
 
-  countOpenApprovals(openApprovals: Trace[], stage: Stage, service?: Service) {
-    return openApprovals.filter(approval => approval == openApprovals.find(a => a.data.stage == stage.stageName && a.data.service == approval.data.service && (!service || a.data.service == service.serviceName))).length;
+  countOpenApprovals(openApprovals: Trace[], project: Project, stage: Stage, service?: Service) {
+    return this.getOpenApprovals(openApprovals, project, stage, service).length;
   }
 
-  getOpenApprovals(openApprovals: Trace[], stage: Stage, service: Service) {
-    return openApprovals.filter(approval => approval.data.stage == stage.stageName && approval.data.service == service.serviceName);
+  getOpenApprovals(openApprovals: Trace[], project: Project, stage: Stage, service?: Service) {
+    return openApprovals.filter(approval => approval.data.project == project.projectName && approval.data.stage == stage.stageName && (!service || approval.data.service == service.serviceName));
```

Comment:
This fixes a bug, that I figured out during testing. When you have the same service (same name) in different projects, and switch between those projects, it was showing you the open approvals also from the other project. Here I added also the filter on projectName, so you see only the approvals in this project, for this service in that stage.

Reply:
Adding the project to uniquely identify a service (and its events) makes total sense.